File: 3074-apple-redistribution-into-boxes/3074-apple-redistribution-into-boxes.java
class Solution {
    public int minimumBoxes(int[] apple, int[] capacity) {
        int sum = 0;
        for(int i : apple){
            sum += i;
        }

        Arrays.sort(capacity);

        int cnt = 0;
        for(int i = capacity.length - 1; i >= 0; i--){
            sum -= capacity[i];
            cnt += 1;
            if(sum <= 0) return cnt;
        }
        return cnt;
    }
}

File: 3074-apple-redistribution-into-boxes/3074-apple-redistribution-into-boxes.py
class Solution:
    def minimumBoxes(self, apple: List[int], capacity: List[int]) -> int:
        apples = sum(apple)
        capacity.sort(reverse = True)

        cnt = 0
        for i in capacity:
            apples -= i
            cnt += 1
            if apples <= 0:
                return cnt
        
        return cnt